### ISSUE TITLE
Install.sh SSLException

### ISSUE THREAD

**Issue by Hawks3** [Darkosto/SevTech-Ages#2280]
## Issue / Bug

While trying to install the server and running the Install.sh script, I came across an issue. The install would claim it installed correctly, but download no files. When looking at the forge-1.12.2-14.23.2.2651-installer.jar.log file, it showed an SSLException saying "the trustAnchors parameter must be non-empty."

## Possible Solution

I found a post that referenced an issue with java cacerts in ubuntu 18.04. I found a java arg that remedied the problem, but having to have this arg is less a fix than a bandage, only aiding the symptoms and not actually solving the issue.
"-Djavax.net.ssl.trustStorePassword=changeit"
## Logs

* Server Install Log: 
`Installing server to current directory
MESSAGE: Data kindly mirrored by Forge at http://MinecraftForge.net/
MESSAGE: Considering minecraft server jar
MESSAGE: Downloading minecraft server version 1.12.2
javax.net.ssl.SSLException: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1944)
	at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1939)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1938)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1508)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:263)
	at net.minecraftforge.installer.DownloadUtils$URLISSupplier.getInput(DownloadUtils.java:457)
	at net.minecraftforge.installer.DownloadUtils$URLISSupplier.getInput(DownloadUtils.java:445)
	at com.google.common.io.ByteStreams$6.openStream(ByteStreams.java:989)
	at com.google.common.io.ByteSource.copyTo(ByteSource.java:202)
	at com.google.common.io.Files.copy(Files.java:382)
	at net.minecraftforge.installer.DownloadUtils.downloadFileEtag(DownloadUtils.java:174)
	at net.minecraftforge.installer.ServerInstall.run(ServerInstall.java:53)
	at net.minecraftforge.installer.InstallerAction.run(InstallerAction.java:43)
	at net.minecraftforge.installer.SimpleInstaller.handleOptions(SimpleInstaller.java:72)
	at net.minecraftforge.installer.SimpleInstaller.main(SimpleInstaller.java:47)
Caused by: javax.net.ssl.SSLException: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
	at sun.security.ssl.Alerts.getSSLException(Alerts.java:208)
	at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1964)
	at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1921)
	at sun.security.ssl.SSLSocketImpl.handleException(SSLSocketImpl.java:1904)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1420)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1397)
	at sun.net.www.protocol.https.HttpsClient.afterConnect(HttpsClient.java:559)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:185)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1564)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
	at sun.net.www.protocol.http.HttpURLConnection.getHeaderField(HttpURLConnection.java:3000)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getHeaderField(HttpsURLConnectionImpl.java:300)
	at net.minecraftforge.installer.DownloadUtils.downloadFileEtag(DownloadUtils.java:163)
	... 4 more
Caused by: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
	at sun.security.validator.PKIXValidator.<init>(PKIXValidator.java:91)
	at sun.security.validator.Validator.getInstance(Validator.java:179)
	at sun.security.ssl.X509TrustManagerImpl.getValidator(X509TrustManagerImpl.java:312)
	at sun.security.ssl.X509TrustManagerImpl.checkTrustedInit(X509TrustManagerImpl.java:171)
	at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:184)
	at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:124)
	at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1596)
	at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:216)
	at sun.security.ssl.Handshaker.processLoop(Handshaker.java:1052)
	at sun.security.ssl.Handshaker.process_record(Handshaker.java:987)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1072)
	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1385)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1413)
	... 12 more
Caused by: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
	at java.security.cert.PKIXParameters.setTrustAnchors(PKIXParameters.java:200)
	at java.security.cert.PKIXParameters.<init>(PKIXParameters.java:120)
	at java.security.cert.PKIXBuilderParameters.<init>(PKIXBuilderParameters.java:104)
	at sun.security.validator.PKIXValidator.<init>(PKIXValidator.java:89)
	... 24 more
Downloading minecraft server failed, invalid e-tag checksum.
Try again, or manually place server jar to skip download.
There was an error during server installation`

## Client Information

* Modpack Version: 3.0.6
* Java Version: 1.8.0_171
* Launcher Used: Twitch

* Memory Allocated: 8GB

* Server/LAN/Single Player: Server.
* Resourcepack Enabled?: None.
* Optifine Installed?: No.

## Server Information
* Java Version: 1.8.0_171
* Operating System: Ubuntu Server 18.04 LTS
* Hoster/Hosting Soultion: Self-Hosted.
* Sponge (Non-Vanilla Forge) Server?:

* Additional Content Installed?: None

**Comment by Hawks3** [comment]
After finally getting the server running, I was unable to connect to it, with the server claiming that the authentication servers were down. After more digging through the log, I was able to find a similar error code to that above. After adding the aforementioned argument to the settings.sh file for the launcher, I was able to connect properly. I believe this is an issue with the java ca certification implementation in Ubuntu 18.04.

**Comment by artdude543** [comment]
1. The SSL from Forge expired now fixed.
2. MC sessions servers were down this morning [MC Status](https://help.mojang.com/)

The  Java CA certification implementation is fine.

**Comment by Hawks3** [comment]
This was not just a last night thing, I have been working on this for about a week now. Even now, it is still not working. I updated everything and tried again just a few moments ago. I got the same errors in the install. And with the workaround I posted I was able to get it to download, and again was not able to connect to the server unless I ran the server with the same parameter. There IS a problem with the Java CA certification implementation in Ubuntu 18.04. This issue has not been resolved, and I ask that it be opened again.
